src/common/utils.ts

```diff
@@ -427,4 +431,29 @@ export const generateSecrets = async (values: Record<string, unknown>): Promise<
   return res
 }
 
+export async function createK8sSecret(name: string, namespace: string, data: Record<string, any>): Promise<void> {
+  const debug: OtomiDebugger = terminal('createK8sSecret')
+  const rawString = JSON.stringify(data)
+  const path = `/tmp/otomi-secret-${namespace}-${name}`
+  writeFileSync(path, rawString)
+  const result = await $`kubectl create secret generic ${name} -n ${namespace} --from-file ${path}`
+  debug.info(result)
+}
+
+export async function getK8sSecret(name: string, namespace: string): Promise<Record<string, any>> {
+  const secretKeyName = `otomi-secret-${namespace}-${name}`
+  const result = await nothrow(
+    $`kubectl get secret ${name} -n ${namespace} -ojsonpath='{.data.${secretKeyName}}' | base64 --decode`,
+  )
+  if (result.exitCode === 0) return JSON.parse(result.stdout)
+  return {}
```

Comment:
Please stray from returning empty objects when we have typescript constructs such as `undefined`.